flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/CommonCorrelate.scala

```diff
@@ -188,7 +189,12 @@ trait CommonCorrelate {
         |getCollector().collect(${crossResultExpr.resultTerm});
         |""".stripMargin
     } else {
-      val filterGenerator = new CodeGenerator(config, false, udtfTypeInfo)
+      val filterGenerator =
+        udtfTypeInfo match {
+          case pt: PojoTypeInfo[_] => new CodeGenerator(config, false, udtfTypeInfo, None,
```

Comment:
I think we don't need a pattern matching here. `pojoFieldMapping` is option anyway so we can pass it directly.